servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Single.java

```diff
@@ -294,6 +294,46 @@ public final Completable flatMapCompletable(Function<? super T, ? extends Comple
         return afterFinally(doFinally);
     }
 
+    /**
+     * Invokes an appropriate terminal signal of the {@link TerminalSignalConsumer}, when any of the following terminal
+     * methods are called:
+     * <ul>
+     *     <li>{@link Subscriber#onSuccess(Object)} - invokes {@link TerminalSignalConsumer#onComplete()}</li>
```

Comment:
Here we have an inconsistency between `Single.whenOnSuccess(Consumer<? super T>)` and `TerminalSignalConsumer.onComplete()` that does not provide `T`. Should we have a different `TerminalSignalConsumer` for `Single`?

Reply:
Ya I think so as `onComplete()` does not match the name (`onSuccess()`) too.

Reply:
If we will have something like `SingleTerminalSignalConsumer`, WDYT about having a `TerminalSignalConsumer` for each async source independently?
- `CompletableTerminalSignalConsumer`
- `SingleTerminalSignalConsumer`
- `PublisherTerminalSignalConsumer`

Also, WDYT about promoting all of them to `servicetalk-concurrent` module where we keep all interfaces for async sources?
- `CompletableSource.TerminalSignalConsumer`
- `SingleSource.TerminalSignalConsumer`
- `PublisherSource.TerminalSignalConsumer`

Reply:
Moving them to `servicetalk-concurrent` module will also help to hide `from(Runnable)` from public API.

Reply:
I am hesitant to push this to `concurrent` module just for reuse of an implementation. This interface is specific to operators and not to the sources.

As we do not expect new callbacks on `Completable` and the terminal callbacks are the same b/w `Publisher` and `Completable`, I think it is less justified to add different interfaces for `Publisher` and `Completable`. I would suggest keeping the current `TerminalSignalConsumer ` as-is, use it for `Publisher` and `Completable` and create a `Single` specific interface which can be an inner class in `concurrent.api.Single`